src/Groups/group_characters.jl

```diff
@@ -212,36 +86,134 @@ mutable struct GAPGroupCharacterTable <: GroupCharacterTable
     end
 end
 
+"""
+    character_table(G::Oscar.GAPGroup, p::Int = 0)
+
+Return the ordinary (if `p == 0`) or `p`-modular character table of the
+finite group `G`.
+If the `p`-modular character table of `G` cannot be computed by GAP
+then `nothing` is returned.
+
+# Examples
+```jldoctest
+julia> character_table( symmetric_group(3) )
+Sym( [ 1 .. 3 ] )
+
+ 2  1  1  .
+ 3  1  .  1
+           
+   1a 2a 3a
+2P 1a 1a 3a
+3P 1a 2a 1a
+           
+χ₁  1 -1  1
+χ₂  2  . -1
+χ₃  1  1  1
+
+
+julia> character_table( symmetric_group(3), 2 )
+Sym( [ 1 .. 3 ] )
+
+ 2  1  .
+ 3  1  1
+        
+   1a 3a
+2P 1a 3a
+3P 1a 1a
+        
+χ₁  1  1
+χ₂  2 -1
+
+
+```
+"""
 function character_table(G::Oscar.GAPGroup, p::Int = 0)
-    tbl = AbstractAlgebra.get_special(G, :character_table)
-    if tbl == nothing
-      tbl = GAP.Globals.CharacterTable(G.X)
-      AbstractAlgebra.set_special(G, :character_table => tbl)
+    tbls = AbstractAlgebra.get_special(G, :character_tables)
+    if tbls == nothing
+      tbls = Dict()
+      AbstractAlgebra.set_special(G, :character_tables => tbls)
     end
 
-    if p != 0
-      isprime(p) || error("p must be 0 or a prime integer")
-      tbl = mod(tbl, p)
+    tbl = get(tbls, p, nothing)
+    if tbl == nothing
```

Comment:
Could also use `get!` plus do block (and then of course also modify the end of the `do` block and the final return
```suggestion
    return get!(tbls, p) do
```